### PR TITLE
Metadata examples

### PR DESCRIPTION
**Pull request by jtourkos**
- Adds helpers for manipulating user metadata (strings <=> bytes)
- Adds UI examples
- Adds a `getNftSubAccountIdsByApp` query method

Feel free to skip the UI files and focus only on the SDK changes.

Merge after https://github.com/radicle-dev/drips-js-sdk/pull/84

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -1,10 +1,62 @@
...
 import { validateAddress, validateDripsReceiverConfig } from './common/validators';
 
 namespace Utils {
+	export namespace UserMetadata {
+		/**
+		 * Creates a user metadata key in the format the Drips protocol expects from the given `string`.
+		 * @param  {string} key The metadata key.
+		 * @returns The metadata key as BytesLike.
+		 */
+		export const keyFromString = (key: string): BytesLike => ethers.utils.formatBytes32String(key);
```
Comment (efstajas) [inline src/utils.ts:14]: Not a blocker, but wondering whether this should validate the max length constraints present on the protocol. Same for `valueFromString`.

File: src/utils.ts [hunk 2]
```diff
@@ -1,10 +1,62 @@
...
+			key: keyFromString(key),
+			value: valueFromString(value)
+		});
+
+		/**
+		 * Returns the given user metadata object in a human readable form.
+		 * @param  {UserMetadataStruct} userMetadata The user metadata.
+		 * @returns An object with the same properties as the `userMetadata` but with `string` properties instead of `BytesLike`.
+		 */
+		export const toHumanReadable = (userMetadata: UserMetadataStruct): { key: string; value: string } => {
```
Comment (efstajas) [inline src/utils.ts:46]: Just regarding of naming, maybe something like `parseMetadataAsString` would be more fitting for this one — given this function would completely mangle any metadata not encoded from a string using the above functions.

File: src/NFTDriver/NFTDriverClient.ts [hunk 3]
```diff
@@ -174,31 +183,41 @@ export default class NFTDriverClient {
...
+			}
+
+			userMetadata.push({ key: dripsConstants.ASSOCIATED_APP_KEY_BYTES, value: associatedApp });
 		}
 
 		const txResponse = await this.#driver.safeMint(transferToAddress, userMetadata);
 
 		const txReceipt = await txResponse.wait();
-		const [transferEvent] = txReceipt.events!;
+		const transferEvent = txReceipt.events!.filter((e) => e.event?.toLowerCase() === 'transfer')[0]!;
```
Comment (efstajas) [inline src/NFTDriver/NFTDriverClient.ts:220]: 🤓 I'm always a bit allergic to `!` in TS — I know that it's practically guaranteed to be safe in this particular scenario, but still think instead asserting `transferEvent` to be truthy, maybe even with a custom error message like `no transfer event could be found`, would make things safer & fail in a clearer way if something super unexpected does happen somehow.

**Review by efstajas — APPROVED** [review]
✌️ Few notes, but nothing blocking.

File: src/utils.ts (hunk 1 above)
Comment (efstajas) [reply src/utils.ts:14]: nevermind, obviously `formatBytes32String` would throw if the input is too long already 👍

File: src/utils.ts (hunk 2 above)
Comment (jtourkos) [reply src/utils.ts:46]: Agree, changed.

File: src/NFTDriver/NFTDriverClient.ts (hunk 3 above)
Comment (jtourkos) [reply src/NFTDriver/NFTDriverClient.ts:232]: Yes, that is a good point. I will fix it here https://github.com/radicle-dev/drips-js-sdk/issues/86 in a separate PR.

**Pull request merged**